packages/dd-trace/src/startup-log.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const mainLogger = require('./log')
+const path = require('path')
+const { inspect } = require('util')
+const tracerVersion = require('../lib/version')
+
+let os
+try {
+  os = require('os')
+} catch (e) {
+  // in browser
+}
+
+const logger = Object.create(mainLogger)
+logger._enabled = process.env.DD_TRACE_STARTUP_LOGS !== '0'
+
+let config
+let plugins = []
+let samplingRules = []
+
+let alreadyRan = false
+
+function getIntegrations () {
+  const integrations = []
+  for (const plugin of plugins) {
+    if (plugin.versions) {
+      try {
+        const version = require(path.join(plugin.name, 'package.json')).version
+        integrations.push(`${plugin.name}@${version}`)
+      } catch (e) {
+        integrations.push(plugin.name)
+      }
+    } else {
+      integrations.push(plugin.name)
+    }
+  }
+  return Array.from(new Set(integrations))
+}
+
+function startupLog (agentError) {
+  if (!config) {
+    return
+  }
+
+  if (alreadyRan) {
+    return
+  }
+
+  alreadyRan = true
+
+  const url = config.url || `http://${config.hostname || 'localhost'}:${config.port}`
```

Comment:
This is not true in every platform, nor every parts of the tracer. It could also become stale by duplicating the logic here. Maybe this information could be split differently?

Reply:
Startup-logging is currently Node.js-specific. Given that, I'm not sure what should change here.

Reply:
The agent has multiple endpoints even for Node. If the semantics of the output entry are well defined, that's probably fine.